"Use picture as" crashes receiving application

### Steps to reproduce
1. Use dot menu for a picture
2. Choose "Use picture as..."
3. Choose any of the possible receiving apps and complete the selection 

### Expected behaviour
- Photo should be set as whatever you chose

### Actual behaviour
- Receiving app crashes

### Environment data
Android version: 11
Device model: emulator
Stock or customized system: stock
Nextcloud app version: 3.18.0
Nextcloud server version: 22
Reverse proxy: no

### Logs
Example with trying to set the photo as a contact picture (Contacts app):
```
2021-11-30 12:35:08.331 3566-3566/com.android.contacts E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.android.contacts, PID: 3566
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/0r1-50324E50/1 flg=0x1 }} to activity {com.android.contacts/com.android.contacts.activities.AttachPhotoActivity}: java.lang.SecurityException: Permission Denial: opening provider com.owncloud.android.providers.DiskLruImageCacheFileProvider from ProcessRecord{7fe371d 3566:com.android.contacts/u0a85} (pid=3566, uid=10085) requires that you obtain access using ACTION_OPEN_DOCUMENT or related APIs
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5015)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:5056)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.SecurityException: Permission Denial: opening provider com.owncloud.android.providers.DiskLruImageCacheFileProvider from ProcessRecord{7fe371d 3566:com.android.contacts/u0a85} (pid=3566, uid=10085) requires that you obtain access using ACTION_OPEN_DOCUMENT or related APIs
        at android.os.Parcel.createExceptionOrNull(Parcel.java:2373)
        at android.os.Parcel.createException(Parcel.java:2357)
        at android.os.Parcel.readException(Parcel.java:2340)
        at android.os.Parcel.readException(Parcel.java:2282)
        at android.app.IActivityManager$Stub$Proxy.getContentProvider(IActivityManager.java:5702)
        at android.app.ActivityThread.acquireProvider(ActivityThread.java:6813)
        at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider(ContextImpl.java:2935)
        at android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:2481)
        at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1967)
        at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1798)
        at android.content.ContentResolver.openInputStream(ContentResolver.java:1475)
        at com.android.contacts.util.ContactPhotoUtils.savePhotoFromUriToUri(ContactPhotoUtils.java:155)
        at com.android.contacts.activities.AttachPhotoActivity.onActivityResult(AttachPhotoActivity.java:189)
        at android.app.Activity.dispatchActivityResult(Activity.java:8304)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5008)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:5056) 
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
     Caused by: android.os.RemoteException: Remote stack trace:
        at com.android.server.am.ActivityManagerService.getContentProviderImpl(ActivityManagerService.java:7155)
        at com.android.server.am.ActivityManagerService.getContentProvider(ActivityManagerService.java:7595)
        at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2381)
        at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2883)
        at android.os.Binder.execTransactInternal(Binder.java:1154)
 ```

--- a/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java
+++ b/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java
@@ -843,6 +843,7 @@ public class FileOperationsHelper {
                 }
 
                 intent.setDataAndType(uri, file.getMimeType());
+                intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
                 fileActivity.startActivityForResult(Intent.createChooser(intent,
                                                                          fileActivity.getString(R.string.set_as)),
                                                     200);

